### PR TITLE
Sampling bias polygon

### PR DESCRIPTION
**Pull request by sunray1**
See #33 - moved polygon commits to a different branch for pull request and reverted commits in original branch

### CODE REVIEW

File: R/apply_polygon_sample_bias.R [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        "You've supplied a {.cls {class(bias_strength)}} vector",
+        "of length {length(bias_strength)}."
+      )
+    ))
+  }
+
+  # 2. check input classes
+  if (!"sf" %in% class(observations)) {
+    cli::cli_abort(c(
+      "{.var observations} must be an sf object",
```
Comment (sangovae) [inline R/apply_polygon_sample_bias.R:77]: add periods at the end of warning messages (guideline)

File: R/apply_polygon_sample_bias.R [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      "x" = "You've supplied a {.cls {class(observations)}} object."
+    ))
+  }
+
+  if (!"sf" %in% class(bias_area)) {
+    cli::cli_abort(c(
+      "{.var bias_area} must be an sf object",
```
Comment (sangovae) [inline R/apply_polygon_sample_bias.R:84]: period

File: R/apply_polygon_sample_bias.R [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+#'
+#' bias_strength <- 2
+#'
+#' apply_polygon_sample_bias(observations_sf, bias_area, bias_strength)
+#'
+apply_polygon_sample_bias <- function(observations, bias_area,
+                                      bias_strength = 1) {
+  require(sf)
+
+  ### Start checks
```
Comment (sangovae) [inline R/apply_polygon_sample_bias.R:62]: Should we have these checks here or in the `sample_observations` function?

**Review by sangovae — APPROVED** [review]
Just a little punctuation remark and question where the checks should belong. Otherwise all fine!

File: R/apply_polygon_sample_bias.R (hunk 3 above)
Comment (sunray1) [reply R/apply_polygon_sample_bias.R:62]: Good point, I'm happy to leave them in for code writing purposes until those checks are implemented in ```sample_observations``` and then they can be removed?

**Comment by sangovae** [comment]
I only realise now, but this function should probably not be exported as it is a helper function?

File: R/apply_polygon_sample_bias.R [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+#' be applied in the biased area (as a multiplier). Above 1, area will be
+#' oversampled. Below 1, area will be undersampled. For example, a value of 50
+#' will result in 50 times more samples within the bias_area than outside.
+#' Conversely, a value of 0.5 will result in half less samples within the
+#' bias_area than outside.
+#'
+#' @returns An sf object with POINT geometry with a bias_weight column
+#' containing the sampling probability based on sampling bias.
+#'
+#' @importFrom cli cli_abort cli_warn
```
Comment (wlangera) [inline R/apply_polygon_sample_bias.R:21]:
```suggestion
#' @importFrom cli cli_abort
```

File: R/apply_polygon_sample_bias.R [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  in_bias_area <- observations %>%
+    sf::st_within(bias_area, sparse = FALSE)
+
+  # Calculate sampling probability based on bias strength
+  bias_weights_outside_polygon <- 1 / (1 + bias_strength)
+  bias_weights_inside_polygon <- bias_strength / (1 + bias_strength)
+
+  #create bias_weight column
+  observations <- observations %>%
+  dplyr::mutate(bias_weight = ifelse(in_bias_area, bias_weights_inside_polygon,
```
Comment (wlangera) [inline R/apply_polygon_sample_bias.R:97]: add indentation
```suggestion
    dplyr::mutate(bias_weight = ifelse(in_bias_area, bias_weights_inside_polygon,
```

Comment (wlangera) [reply R/apply_polygon_sample_bias.R:97]: maybe check if the indentation of the line under this is also fine

**Review by wlangera — APPROVED** [review]
Great work. I have some small coding style remarks.

I have also pulled all updates from the main.
Still 1 check is failing, but I think this is because there are no unit tests for this function? @PietrH is this correct?

**Comment by sunray1** [comment]
Should we build unit tests firsts or go ahead and merge? @wlangera @PietrH

**Comment by wlangera** [comment]
> Should we build unit tests firsts or go ahead and merge? @wlangera @PietrH

@sunray1 you can make the unit tests :) so you can continue

**Pull request merged**